api/spec/services/validator/stylesheet_spec.rb

```diff
@@ -0,0 +1,159 @@
+    expect(compact(results)).to eq compact(valid)
+  end
+
+  it "should not allow a * selector" do
+    invalid = "* { font-weight: bold; }"
+    valid = ""
+    results = validator.validate(invalid)
+    expect(compact(results)).to eq compact(valid)
+  end
+
+  it "should not allow a @font-family selector" do
+    invalid = "@font-face { font-family: \"Gallery\"; }"
+    valid = ""
+    results = validator.validate(invalid)
+    expect(compact(results)).to eq compact(valid)
+  end
+
+  it "should properly scope multiple selectors" do
+    invalid = "span, p { font-weight: bold; }"
+    valid = "#{scope_selector} span { font-weight: bold; }
+             #{scope_selector} p { font-weight: bold; }"
+    results = validator.validate(invalid)
+    expect(compact(results)).to eq compact(valid)
+  end
+
+  describe "with complex selectors" do
+
+    invalid_test_cases = [
+      ["element", "* { foo: bar; }"],
+      ["general sibling", "div ~ * { foo: bar; }"],
+      ["adjacent sibling", "div + * { foo: bar; }"],
+      ["child combinator", "div > * { foo: bar; }"],
+      ["descendant combinator", "div * { foo: bar; }"],
+      ["negation psuedo class", "*:not(foo) { foo: bar; }"],
+      ["after psuedo-element", "*:after { foo: bar; }"],
+      ["before psuedo-element", "*:before { foo: bar; }"],
+      ["first-letter psuedo-element", "*:first-letter { foo: bar; }"],
+      ["first-line psuedo-element", "*:first-line { foo: bar; }"],
+      ["checked psuedo-class", "*:checked { foo: bar; }"],
+      ["disabled psuedo-class", "*:disabled { foo: bar; }"],
+      ["enabled psuedo-class", "*:enabled { foo: bar; }"],
+      ["lang psuedo-class", "*:lang { foo: bar; }"],
+      ["target psuedo-class", "*:target { foo: bar; }"],
+      ["focus psuedo-class", "*:focus { foo: bar; }"],
+      ["hover psuedo-class", "*:hover { foo: bar; }"],
+      ["active psuedo-class", "*:active { foo: bar; }"],
+      ["visited psuedo-class", "*:visited { foo: bar; }"],
+      ["link psuedo-class", "*:link { foo: bar; }"],
+      ["empty psuedo-class", "*:empty { foo: bar; }"],
+      ["only-of-type psuedo-class", "*:only-of-type { foo: bar; }"],
+      ["only-child psuedo-class", "*:only-child { foo: bar; }"],
+      ["last-of-type psuedo-class", "*:last-of-type { foo: bar; }"],
+      ["first-of-type psuedo-class", "*:first-of-type { foo: bar; }"],
+      ["last-child psuedo-class", "*:last-child { foo: bar; }"],
+      ["first-child psuedo-class", "*:first-child { foo: bar; }"],
+      ["nth-of-type(1) psuedo-class", "*:nth-of-type(1) { foo: bar; }"],
+      ["nth-last-of-type(1) psuedo-class", "*:nth-last-of-type(1) { foo: bar; }"],
+      ["nth-child(1) psuedo-class", "*:nth-child(1) { foo: bar; }"],
+      ["root psuedo-class", "*:root { foo: bar; }"],
+      ["attribute selector", "*[foo=\"bar\"] { foo: bar; }"]
+    ]
+    valid_test_cases = invalid_test_cases.map { |test_case| [test_case[0], test_case[1].sub("*", "div")]}
```

Comment:
Metrics/LineLength: Line is too long. [105/90]<br>Style/SpaceInsideBlockBraces: Space missing inside }.